### DISCUSSION TITLE
How to Hardcode Setoption?

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#19074] [Q&A] [answered]
Hello, Just used TuyaConvert to flash Tasmota 13.0.0 on a Wipro RGB Bulb successfully. As per template on blakadder.com I need to change Setoption92 to 1 or ON. I have already done that using the console. However, I would like to generate a custom binary with the template and this "setoption92" 1 hard coded. Can I use backlog command after the user template in the binary as follows:
```
#define MODULE            USER_MODULE
#define USER_TEMPLATE "{\"NAME\":\"Wipro-RGB-NS9001\",\"GPIO\":[0,0,0,0,416,419,0,0,417,420,418,0,0,0],\"FLAG\":0,\"BASE\":18}"
#define USER_BACKLOG "Setoption92 1"
#define EMULATION         EMUL_HUE
#define USE_EMULATION_HUE
#define DEVICE_NAME       "Wipro RGB Bulb"  
#define FRIENDLY_NAME     "Wipro RGB Bulb 1"
#define USE_TASMOTA_DISCOVERY
```
If not then what will be the best method to hardcode "Setoption92 1"? Thanks in advance.

![Screenshot (259)](https://github.com/arendst/Tasmota/assets/69470381/32560494-a4d2-4d64-88f8-2621b5ffaf71)

**Comment by sfromis** [comment, marked as answer]
While you *can* use `USER_BACKLOG`, the most direct would instead be to use the specific define already available for the purpose.
`#define LIGHT_PWM_CT_MODE      false             // [SetOption92] Set PWM Mode ......`

You could also include the command in the template, instead of using `USER_BACKLOG`
`"CMND":"SetOption92 1"`

**Comment by NsinghP** [comment]
> While you _can_ use `USER_BACKLOG`, the most direct would instead be to use the specific define already available for the purpose. `#define LIGHT_PWM_CT_MODE false // [SetOption92] Set PWM Mode ......`
> 
> You could also include the command in the template, instead of using `USER_BACKLOG` `"CMND":"SetOption92 1"`

Thankyou for a detailed reply. I am simply going to use:

`#define LIGHT_PWM_CT_MODE      true`

However, I am a bit confused about using the template method! If I were to do that, would the template look something like this:

`#define USER_TEMPLATE "{\"CMND\":\"SetOption92 1\",\"NAME\":\"Wipro-RGB-NS9001\",\"GPIO\":[0,0,0,0,416,419,0,0,417,420,418,0,0,0],\"FLAG\":0,\"BASE\":18}"`

**Reply by sfromis** [reply]
Yes, something like that. Usually the `"CMND":` is put last, but I'm not sure of this making any difference. Any command sequence included in the template is executed when the template is applied, and then the template is stored without this field.
